lib/music_library_controller.rb

```diff
@@ -0,0 +1,86 @@
+class MusicLibraryController
+  attr_accessor :path
+
+  def initialize(path = "./db/mp3s")
+    @path = path
+    MusicImporter.new(path).import
+  end
+
+  def call
+    input = ""
+
+    while input != "exit"
+    puts "Welcome to your music library!"
+    puts "To list all of your songs, enter 'list songs'."
+    puts "To list all of the artists in your library, enter 'list artists'."
+    puts "To list all of the genres in your library, enter 'list genres'."
+    puts "To list all of the songs by a particular artist, enter 'list artist'."
+    puts "To list all of the songs of a particular genre, enter 'list genre'."
+    puts "To play a song, enter 'play song'."
+    puts "To quit, type 'exit'."
+    puts "What would you like to do?"
+
+      input = gets.strip
+
+    list_songs if input == "list songs"
+    list_artists if input == "list artists"
+    list_genres if input == "list genres"
+    list_songs_by_artist if input == "list artist"
+    list_songs_by_genre if input == "list genre"
+    play_song if input == "play song"
+    end
+  end
+
+  def list_songs
+    Song.all.sort {|a, b| a.name <=> b.name}.each.with_index(1) do |song, i|
+      puts "#{i}. #{song.artist.name} - #{song.name} - #{song.genre.name}"
+    end
+  end
+
+  def list_artists
+    Artist.all.sort { |a, b| a.name <=> b.name }.each.with_index(1) do |artist, i|
+      puts "#{i}. #{artist.name}"
+    end
+  end
+
+  def list_genres
+    Genre.all.sort {|a, b| a.name <=> b.name}.each.with_index(1) do |genre, i|
```

Comment:
Using `sort_by` for all your list methods would make it syntactically easier to read.

Also `each.with_index` (2 methods) can be replaced by `each_with_index` (1 method). This would also reduce a little complexity.